cluster/src/dunit/scala/io/snappydata/cluster/ConcurrentQueryRoutingDUnitSecureTest.scala

```diff
@@ -0,0 +1,465 @@
+/*
+ * Copyright (c) 2016 SnappyData, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you
+ * may not use this file except in compliance with the License. You
+ * may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License. See accompanying
+ * LICENSE file.
+ */
+
+package io.snappydata.cluster
+
+import java.sql.{BatchUpdateException, Connection, DriverManager, ResultSet, SQLException}
+import java.util.concurrent.atomic.AtomicInteger
+
+import io.snappydata.test.dunit.AvailablePortHelper
+
+import org.apache.spark.Logging
+import org.apache.spark.sql.collection.Utils
+
+object ConcurrentQueryRoutingDUnitSecureTest{
+  val adminUser: String = "gemfire7"
+}
+
+class ConcurrentQueryRoutingDUnitSecureTest(val s: String)
+    extends ClusterManagerLDAPTestBase(s,
+      ConcurrentQueryRoutingDUnitSecureTest.adminUser) with Logging {
+
+  override def setUp(): Unit = {
+    super.setUp()
+  }
+
+  override def tearDown2(): Unit = {
+    super.tearDown2()
+  }
+
+  def netConnection(netPort: Int, user: String, pass: String): Connection = {
+    val driver = "io.snappydata.jdbc.ClientDriver"
+    Utils.classForName(driver).newInstance
+    val url: String = "jdbc:snappydata://localhost:" + netPort + "/"
+    DriverManager.getConnection(url, user, pass)
+  }
+
+  def insertRows1(numRows: Int, serverHostPort: Int, tableName: String,
```

Comment:
It will really help if you can add a comment on top of the method as to what is 1, 2, 3 and 4 or make the name changes such that insert style differences can be easily understood.
1, 3 and 2, 4 look very similar. Spot the difference :-)